apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -344,20 +344,93 @@ func (s *SecretsManagerAPI) GetSecretContentInfo(args params.GetSecretContentArg
 	return result, nil
 }
 
-func (s *SecretsManagerAPI) getSecretContent(arg params.GetSecretContentArg) (*secrets.ContentParams, error) {
-	uri, err := coresecrets.ParseURI(arg.URI)
+func (s *SecretsManagerAPI) getOwnerSecretMetadata(uri *coresecrets.URI, label string) (*coresecrets.SecretMetadata, error) {
+	if uri == nil {
+		var err error
+		uri, err = s.secretsBackend.GetURIBySecretLabel(label, s.authTag)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+
+	md, err := s.secretsBackend.GetSecret(uri)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	if !s.canRead(uri, s.authTag) {
-		return nil, apiservererrors.ErrPerm
+	if s.authTag.String() != md.OwnerTag {
+		// Invisible for non owner users.
+		return nil, errors.NotFoundf("secret %s, label %q", uri, label)
+	}
+	return md, nil
+}
+
+func (s *SecretsManagerAPI) getSecretContent(arg params.GetSecretContentArg) (*secrets.ContentParams, error) {
+	// Only the owner can access secrets via the secret label (Note: the leader unit is not the owner of the application secrets).
+	// Non owners have to use secret URI or consumer label.
```

Comment:
// Only the owner can access secrets via the secret metadata label added by the owner.
// (Note: the leader unit is not the owner of the application secrets).
// Consumers get to use their own label.
// Both owners and consumers can also just use the secret URI.